modules/oaiharvest/lib/oai_harvest_daemon.py

```diff
@@ -572,36 +572,47 @@ def remove_duplicates(harvested_file_list):
     """
     Go through a list of harvested files and remove any duplicate records.
     """
+    from copy import deepcopy
+    import lxml.etree as etree
```

Comment:
Indeed we'd need it fixed in 1.1 till 1.2 is released and we can upgrade. In fact we badly need it and applied our fix locally, via our fork.

Concerning *keeping old behaviour*: we should consider then to find a way to get a more meaningful reporting if the problem occurs. Note that we stumbled upon the error with a source we used for quite some time (cordis) due to changes they made on there end. Old behaviour doesn't even produce an error, we just silently didn't get any datasets though we adopted all XSLTs and stuff. Given that we don't expect an update from that source every day it took quite some time to see the problem. It took a while to even fiddle out that there was indeed a problem on the invenio side let alone where the problem lay (given we got valid XML).

Given that I'm not sure that adding lxml to the requirements isn't the cheaper price to pay for "the admin out there"(tm).